src/broker/service.c

```
@@ -161,7 +161,7 @@ int service_add (struct service_switch *sh, const char *name,
     svc->cb = cb;
     svc->cb_arg = arg;
     if (zhash_insert (sh->services, name, svc) < 0) {
-        errno = ENOMEM;
+        errno = EEXIST;
```

same here, ignore the error